src/garage/tf/algos/npo2.py

```diff
@@ -0,0 +1,562 @@
+            'no_entropy'. The type of entropy method to use. 'max' adds the
+            dense entropy to the reward for each time step. 'regularized' adds
+            the mean entropy to the surrogate objective. See
+            https://arxiv.org/abs/1805.00909 for more details.
+        flatten_input (bool): Whether to flatten input along the observation
+            dimension. If True, for example, an observation with shape (2, 4)
+            will be flattened to 8.
+        name (str): The name of the algorithm.
+
+    Note:
+        sane defaults for entropy configuration:
+            - entropy_method='max', center_adv=False, stop_gradient=True
+              (center_adv normalizes the advantages tensor, which will
+              significantly alleviate the effect of entropy. It is also
+              recommended to turn off entropy gradient so that the agent
+              will focus on high-entropy actions instead of increasing the
+              variance of the distribution.)
+            - entropy_method='regularized', stop_gradient=False,
+              use_neg_logli_entropy=False
+
+    """
+
+    def __init__(self,
+                 env_spec,
+                 policy,
+                 baseline,
+                 scope=None,
+                 max_path_length=500,
+                 discount=0.99,
+                 gae_lambda=1,
+                 center_adv=True,
+                 positive_adv=False,
+                 fixed_horizon=False,
+                 pg_loss='surrogate',
+                 lr_clip_range=0.01,
+                 max_kl_step=0.01,
+                 optimizer=None,
+                 optimizer_args=None,
+                 policy_ent_coeff=0.0,
+                 use_softplus_entropy=False,
+                 use_neg_logli_entropy=False,
+                 stop_entropy_gradient=False,
+                 entropy_method='no_entropy',
+                 flatten_input=True,
+                 name='NPO'):
+        self._name = name
+        self._name_scope = tf.name_scope(self._name)
+        self._old_policy = policy.clone('old_policy')
+        self._use_softplus_entropy = use_softplus_entropy
+        self._use_neg_logli_entropy = use_neg_logli_entropy
+        self._stop_entropy_gradient = stop_entropy_gradient
+        self._pg_loss = pg_loss
+        if optimizer is None:
+            if optimizer_args is None:
+                optimizer_args = dict()
+            optimizer = LbfgsOptimizer
+
+        self._check_entropy_configuration(entropy_method, center_adv,
+                                          stop_entropy_gradient,
+                                          use_neg_logli_entropy,
+                                          policy_ent_coeff)
+
+        if pg_loss not in ['vanilla', 'surrogate', 'surrogate_clip']:
+            raise ValueError('Invalid pg_loss')
+
+        with self._name_scope:
+            self._optimizer = optimizer(**optimizer_args)
+            self._lr_clip_range = float(lr_clip_range)
```

Comment:
i don't think the following 3 need to be in a name scope...

Reply:
Sure